microdnf-plugins/builddep_plugin/builddep.cpp

```diff
@@ -0,0 +1,263 @@
+
+namespace microdnf {
+
+
+using namespace libdnf::cli;
+
+BuildDepCommand::BuildDepCommand(Command & parent) : Command(parent, "builddep") {
+    auto & ctx = static_cast<Context &>(get_session());
+    auto & parser = ctx.get_argument_parser();
+
+    auto & cmd = *get_argument_parser_command();
+    cmd.set_short_description("Install build dependencies for package or spec file");
+
+    auto specs = parser.add_new_positional_arg("specs", ArgumentParser::PositionalArg::AT_LEAST_ONE, nullptr, nullptr);
+    specs->set_short_description("List of specifications. Accepts *.spec / *.src.rpm files or package name.");
+    specs->set_parse_hook_func(
+        [this]([[maybe_unused]] ArgumentParser::PositionalArg * arg, int argc, const char * const argv[]) {
+            parse_builddep_specs(argc, argv);
+            return true;
+        });
+    specs->set_complete_hook_func([&ctx](const char * arg) {
+        return match_specs(ctx, arg, false, true, true, false, ".*\\.(spec|src\\.rpm|nosrc\\.rpm)");
+    });
+    cmd.register_positional_arg(specs);
+
+    auto skip_unavailable = parser.add_new_named_arg("skip-unavailable");
+    skip_unavailable->set_long_name("skip-unavailable");
+    skip_unavailable->set_short_description("Skip build dependencies not available in repositories");
+    skip_unavailable->set_const_value("true");
+    skip_unavailable->link_value(&skip_unavailable_option);
+    cmd.register_named_arg(skip_unavailable);
+
+    auto defs = parser.add_new_named_arg("rpm_macros");
+    defs->set_short_name('D');
+    defs->set_long_name("define");
+    defs->set_has_value(true);
+    defs->set_arg_value_help("\"macro expr\"");
+    defs->set_short_description("Define the RPM macro named \"macro\" to the value \"expr\" when parsing spec files");
+    defs->set_parse_hook_func(
+        [this](
+            [[maybe_unused]] ArgumentParser::NamedArg * arg, [[maybe_unused]] const char * option, const char * value) {
+            auto split = libdnf::utils::string::split(value, " ", 2);
+            if (split.size() != 2) {
+                throw libdnf::cli::ArgumentParserError(
+                    "Invalid value for macro definition \"{}\". \"macro expr\" format expected.", value);
+            }
+            rpm_macros.emplace_back(std::move(split[0]), std::move(split[1]));
+            return true;
+        });
+    cmd.register_named_arg(defs);
+}
+
+
+void BuildDepCommand::parse_builddep_specs(int specs_count, const char * const specs[]) {
+    const std::string_view ext_spec(".spec");
+    const std::string_view ext_srpm(".src.rpm");
+    const std::string_view ext_nosrpm(".nosrc.rpm");
+    std::set<std::string> unique_items;
+    for (int i = 0; i < specs_count; ++i) {
+        const std::string_view spec(specs[i]);
+        if (auto [it, inserted] = unique_items.emplace(spec); inserted) {
+            // TODO(mblaha): download remote URLs to temporary location + remove them afterwards
+            if (spec.ends_with(ext_spec)) {
```

Comment:
The `string_view::ends_with` method has been supported since C++20. It is used in multiple places.
 I'm not sure if we no longer need C++17 compatibility.
We use our `libdnf::utils::string::ends_with` function in libdnf. And you're using it in `RepoSack :: enable_source_repos.`

Reply:
Yeah, I know. But there already is `set(CMAKE_CXX_STANDARD 20)` so I went ahead and used it.

Reply:
After discussion, we came to the conclusion that we no longer need compatibility with C++17. We will use C++20.
So we can keep this code.